src/Main.java

```diff
@@ -1,42 +1,106 @@
+                } else {
+                    System.out.println(ongeldig);
+                }
+            } else {
+                System.out.println(ongeldig);
+            }
+        }
+
+
+        HashSet<Integer> secretNumber = randomNumberGenerator();
+        String stringNumber =  setToStringConverter(secretNumber);
+        System.out.println(stringNumber);
+        feedback(stringNumber);
 
     }
 
+    public static HashSet<Integer> randomNumberGenerator() {
+        HashSet<Integer> randomNumbers = new HashSet<>();
+        Random random = new Random();
+        int i = 0;
+        while(i <= 4) {
+            randomNumbers.add(random.nextInt(10));
+            i++;
+        }
+        return randomNumbers;
+    }
+
+    public static String setToStringConverter(HashSet<Integer> randomNumbers) {
+        StringBuilder randomNumbersString = new StringBuilder();
+        for (Integer randomNumber : randomNumbers) {
+            randomNumbersString.append(randomNumber.toString());
+        }
+        return randomNumbersString.toString();
+    }
+
     /*
      Deze methode is voor de bonus opdracht.
      */
-    public static void feedback(String stringnumber) {
+    public static void feedback(String stringNumber) {
+        boolean play = true;
         Scanner scanner = new Scanner(System.in);
-        StringBuilder feedback = new StringBuilder();
-        System.out.println("take a guess");
-        String guess = scanner.nextLine();
-        if (Objects.equals(guess, stringnumber)) {
-            System.out.println("gefeliciteerd je hebt het goed");
-        } else {
-            for (int i = 0; i < 4; i++) {
-                if (guess.substring(i, i + 1).equals(stringnumber.substring(i, i + 1))) {
-                    feedback.append("+");
-                } else if (stringnumber.contains(guess.substring(i, i + 1))) {
-                    feedback.append("0");
-                } else {
-                    feedback.append("X");
+
+        while (play) {
+            StringBuilder feedback = new StringBuilder();
+            System.out.println("take a guess");
+            String guess = scanner.nextLine();
+            if (Objects.equals(guess, stringNumber)) {
+                System.out.println("gefeliciteerd je hebt het goed");
+                play = false;
+            } else {
+                for (int i = 0; i < 4; i++) {
+                    if (guess.substring(i, i + 1).equals(stringNumber.substring(i, i + 1))) {
+                        feedback.append("+");
+                    } else if (stringNumber.contains(guess.substring(i, i + 1))) {
+                        feedback.append("0");
+                    } else {
+                        feedback.append("X");
+                    }
                 }
             }
+            System.out.println(feedback.toString());
+            System.out.println("Do you want to guess again? y/n");
```

Comment:
loop werkt niet want hij vraagt na eenmaal een gok gewaagd te hebben of ik nog een keer wilt spelen.